lib/external/sentry.js

```diff
@@ -1,7 +1,80 @@
 const { inspect } = require('util');
 const { isBlank } = require('../util/util');
 
-const sensitiveEndpoints = ['/users/reset/verify', '/users/(.*?)/password'];
+
+// Endpoints where query strings are sensitive and should be removed
+const sensitiveEndpoints = [
+  ['GET', '/users[^/]'], // ?q=<part of name or email>
+  ['GET', '/projects/([^/]*)/forms/([^/]*)/submissions.csv'], // ?key=passphrase for managed encryption
+  ['GET', '/projects/([^/]*)/forms/([^/]*)/draft/submissions.csv'] // ?key=passphrase for managed encryption
+];
+
+const isSensitiveEndpoint = (request) => {
+  for (const [method, endpoint] of sensitiveEndpoints) {
+    if (request.method === method && request.url.match(endpoint)) {
+      return true;
+    }
+  }
+  return false;
+};
+
+const filterTokenFromUrl = (url) => {
+  // remove user token from any URL
+  if (url.match('/v1/key/(.*?)')) {
+    return url.replace(/v1\/key\/([^/]*)/, 'v1/key/[FILTERED]');
+  }
+
+  // remove draft token from any URL
+  if (url.match('/v1/test/(.*?)')) {
+    return url.replace(/v1\/test\/([^/]*)/, 'v1/test/[FILTERED]');
+  }
+
+  // return original URL if there are no tokens
+  return url;
+};
```

Comment:
Ah, I hadn't seen this before because I hadn't tried using enketo with public links. I've now had a chance to look into it, and into the middleware that transforms it into the `key` url. There is now new code and tests (including ones I saw when I manually broke the middleware) to reflect this case.

I can't easily remove just this specific query parameter, but I added these URLs that enketo sends `st` with to the list of sensitive endpoints that will have all their query parameters stripped.

I also saw this `st` param getting captured in the request header `referer` field, so I stripped that out of all requests, too.